jobs/generation/PRTestGenerator.groovy

```diff
@@ -0,0 +1,482 @@
+
+                // Always archive so that the flexible publishing
+                // handles pass/fail
+                onlyIfSuccessful(false)
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+class ProjectCoreclr extends Project {
+  def setupJob(def newJob, String branch, String scenario, boolean isPR, String architecture, String configuration, String os) {
+    Utilities.standardJobSetup(newJob, name, true, "*/${branch}")
+
+    addPRTriggers(newJob, branch, architecture, os, configuration, scenario)
+    setJobTimeout(newJob, architecture, configuration, scenario)
+
+    newJob.with {
+      steps {
+        copyArtifacts("${getJobFolderName('dotnet/corefx', branch)}/${getJobName(configuration != 'Checked' ? configuration : 'Release', architecture, os, scenario)}") {
+          includePatterns("${getCorefxBinPath(branch)}/build.tar.gz")
+          buildSelector {
+            latestSuccessful(true)
+          }
+        }
+      }
+    }
+
+    if (branch == 'release/2.0.0') {
+      String patch = 'coreclr-v2.0.0-tests.patch'
+      String newJobName = getJobName(configuration, architecture, os, scenario)
+      String folderName = getJobFolderName(branch)
+      String projectDir = Utilities.getFullJobName(name, newJobName, isPR, folderName)
+      Utilities.getPatch(newJob, patch)
+      Utilities.applyPatch(newJob, patch)
+    }
+
+    List<String> buildCommands = calculateBuildCommands(newJob, scenario, branch, architecture, configuration, os, configuration != 'Checked')
+
+    newJob.with {
+      steps {
+        buildCommands.each { buildCommand ->
+          shell(buildCommand)
+        }
+      }
+    }
+  }
+
+  boolean shouldGenerateJob(String scenario, boolean isPR, String architecture, String configuration, String os) {
+    // Generate only PR jobs
+    return isPR;
+  }
+
+  static String getDotnetPath(String branch)
+  {
+    switch (branch) {
+      case 'release/2.0.0':
+      case 'release/2.1':
+      case 'release/2.2':
+        return './Tools/dotnetcli/dotnet'
+      case 'release/3.0':
+      case 'master':
+        return './.dotnet/dotnet'
+      default:
+        assert false : "Unknown branch: '${branch}'"
+        break
+    }
+  }
+
+  static List<String> calculateBuildCommands(def newJob, String scenario, String branch, String architecture,
+                                             String configuration, String os, boolean isBuildOnly) {
+    List<String> buildCommands = []
+    String osGroup = getOSGroup(os)
+    String lowerConfiguration = configuration.toLowerCase()
+
+    // Calculate the build steps, archival, and xunit results
+    switch (os) {
+      case 'Tizen':
+        switch (architecture) {
+          case 'armel':
+            String linuxCodeName = os.toLowerCase()
+
+            assert scenario == 'normal' : "Unsupported scenario: '${scenario}'"
+
+            String xUnitTestBinBase = "./bin/tests/Windows_NT.x64.${configuration}"
```

Comment:
@gbalykov do we use Windows tests?

Reply:
MS used Windows tests in CI, this job compiles tests on Linux but [arm32_ci_test.sh](https://github.com/dotnet/coreclr/blob/234f1b6b238a9faa860a0bd1253d4baabdb35759/tests/scripts/arm32_ci_test.sh#L108) expects that tests are built for Windows so we use this path.